lib/WUI/nhttp/headers.cpp

```diff
@@ -139,4 +147,33 @@ ContentType guess_content_by_ext(const char *filename) {
     return ContentType::ApplicationOctetStream;
 }
 
+uint32_t compute_etag(const struct stat &stat) {
+    /*
+     * We need something that hopefully changes whenever someone replaces a
+     * file with a different file but the same name. Hashing a bit of metadata
+     * seems like a good candidate.
+     */
+    mbedtls_sha256_context ctx;
+    mbedtls_sha256_init(&ctx);
+    mbedtls_sha256_starts_ret(&ctx, false);
+    mbedtls_sha256_update_ret(&ctx, (const uint8_t *)&stat.st_ino, sizeof stat.st_ino);
```

Comment:
Have you checked, that our POSIX filesystem layer is actually able to provide some meaningful inode number from the underlying FATfs and littleFS?